File: laboratorka-5.py
# Вариант 23 => 1 и 14
# Задание 1
num = int(input('Введите натуральное число: '))
while num > 0:
    print(num % 10)
    num //= 10
print()

# Задание 14
num = int(input('Введите натуральное число: '))
result = 0
while num > 0:
    if num % 10 < 5:
        result = result + 1
    num //= 10
print('Количество цифр меньших 5 =', result)

File: laboratorka_5.cpp
#include <iostream>
#include <cmath>
#include <math.h>
using namespace std;

long double fact(int N)
{
    if (N < 0) return 0;
    if (N == 0) return 1;
    return N * fact(N - 1);
}

int main() {
	long double s = 0, eps = 0.001, term = eps, x;
	int counter = 0;
	
	do{
		cout << "Input -1000 < x < 1000: ";
		cin >> x;
	} while(x >= 1000 || x <= -1000);
	
	cout << "e^x = " << exp(x) << endl;
	
	while(abs(term) >= eps){
		term = pow(x, counter) / fact(counter);
		s += term;
		counter++;
	}
	
	cout << "s = " << s << endl;
	return 0;
}
